src/robot/libraries/Telnet.py

```diff
@@ -430,6 +444,32 @@ def _get_connection(self, *args):
         """Can be overridden to use a custom connection."""
         return TelnetConnection(*args)
 
+    def set_connection_timeout(self, connection_timeout):
+        """Sets the timeout used when opening a new connection.
+
+        Open connection operations use this timeout and fail if the
+        connection is not established before this timeout expires.
+
+        The ``connection_timeout`` must be given in `time string format`.
+        The old connection timeout is returned and can be used to restore
+        the timeout later.
+
+        Example:
+        | ${old} =          | `Set Connection Timeout` | 2 minutes 30 seconds |
+        | `Open Connection` | ${host}
+
+        See `Configuration` section for more information about global and
+        connection specific configuration.
+        """
+        old = self._connection_timeout
+        self._set_connection_timeout(connection_timeout)
+        return secs_to_timestr(old) if old is not None else None
```

Comment:
I think it is enough. I added this to maintain consistency with the "normal" timeout keywords. I'll remove it.